### PR TITLE
Add environment variable to set minimum TLS version on server tls config

### PR DESCRIPTION
**Pull request by denkyl08**
This will allow folks to block usage of TLS 1.0 and TLS1.1 which have known vulnerabilities.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -517,6 +518,25 @@ func getEnvBool(log logrus.FieldLogger, name string) bool {
...
+	minVersion := map[string]uint16{
+		"1.0": tls.VersionTLS10,
+		"1.1": tls.VersionTLS11,
+		"1.2": tls.VersionTLS12,
+		"1.3": tls.VersionTLS13,
+	}
+	version, ok := minVersion[v]
+	if !ok {
+		log.WithError(fmt.Errorf("unknown tls version: %s", v)).Errorf("while parsing '%s' as an min tls version, defaulting to 1.0", name)
+		return tls.VersionTLS10
```
Comment (Baliedge) [inline config.go:535]: The default ought to be the most secure option, 1.3.

**Review by Baliedge — CHANGES_REQUESTED** [review]
I like this security-minded update.  Just a small behavioral change I recommended.  Thanks for the contribution!

**Review by Baliedge — CHANGES_REQUESTED** [review]
Code looks great.  Just get the checks to pass and I'll merge.

**Comment by denkyl08** [comment]
@Baliedge  I am not able to reproduce the test failure locally. Could you re-run these jobs? Also, I'm not sure why the first job  was cancelled, do you know?

**Review by Baliedge — APPROVED** [review]
That turned out to be a pre-existing test issue that'll need to be investigated later.  Approved.

**Pull request merged**